task/task_smoke_test.go

```diff
@@ -19,19 +19,62 @@ import (
 	"terraform-provider-iterative/task/common"
 )
 
+type testTarget struct {
+	machine string
+	image   string
+	script  string
+}
+
+var testTargets = map[string]testTarget{
+	// default test target spins up small instances and runs lightweight scripts on them.
+	"default": testTarget{
+		machine: "m",
+		image:   "ubuntu",
+		script: `
+#!/bin/sh -e
+mkdir --parents cache output
+touch cache/file
+echo "$ENVIRONMENT_VARIABLE_DATA" | tee --append output/file
+sleep 60
+cat output/file
+`[1:],
+	},
+	"gpu": testTarget{
+		machine: "m+t4",
+		image:   "nvidia",
+		script: `
+#!/bin/sh -e
+nvidia-smi
+mkdir --parents cache output
+touch cache/file
+echo "$ENVIRONMENT_VARIABLE_DATA" | tee --append output/file
+sleep 60
+cat output/file
+`[1:],
+	},
+}
```

Comment:
I don't quite like the idea of maintaining two copies of the script just to skip a line. Instead of parameterizing `script` you could parameterize `environment` and set `TEST_GPU=true` so the script can branch accordingly.[^1]

[^1]: Depending on how purist you are, `$TEST_GPU && nvidia-smi` or `printenv TEST_GPU && nvidia-smi` can be chosen over `test -z "$TEST_GPU" && nvidia-smi` or a dozen other combinations.

Reply:
ok, updated